### PR TITLE
Add storm and update http api docs (SYN-4030, SYN-4041, SYN-4044)

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Add storm specific docs for opts
- Add storm specific docs for message types
- Update http api docs to include requests based examples
- Update http api docs to include api/v1/storm and api/v1/storm/call examples.
- Include `mesg` args in s_exc.IsReadOnly exceptions.

### CODE REVIEW

File: docs/synapse/httpapi.rst [hunk 1]
```diff
@@ -0,0 +1,804 @@
...
+
+While not in "insecure" mode, most Synapse HTTP APIs require an authenticated user.
+HTTP API endpoints requiring authentication may be accessed using either HTTP Basic
+authentication via the HTTP "Authorization" header or as part of an authenticated
+session.  For more information on configuring users/roles see TODO.
+
+To create and use an authenticated session, the HTTP client library must support
+cookies.
+
+/api/v1/login
```
Comment (vEpiphyte) [inline docs/synapse/httpapi.rst:49]: This section is updated

File: docs/synapse/httpapi.rst [hunk 2]
```diff
@@ -0,0 +1,804 @@
...
+
+::
+
+    {"status": "err": "code": "ErrCodeString", "mesg": "A human friendly message."}
+
+Streaming HTTP API endpoints, such as the interface provided to retrieve nodes
+from a Synapse Cortex, provide JSON results via HTTP chunked encoding where each
+chunk is a single result.
+
+The client example code in these docs is given with the Python "aiohttp" and "requests"
```
Comment (vEpiphyte) [inline docs/synapse/httpapi.rst:33]: This section is updated

File: docs/synapse/httpapi.rst [hunk 3]
```diff
@@ -0,0 +1,804 @@
...
+
+        Each ``<node>`` is expected to be in packed tuple form::
+
+            [ [<formname>, <formvalu>], {...} ]
+
+    *Returns*
+        The API returns ``{"status": "ok", "result": null}`` on success and any failures
+        are returned using the previously mentioned REST API convention.
+
+/api/v1/storm
```
Comment (vEpiphyte) [inline docs/synapse/httpapi.rst:313]: This section is updated

File: docs/synapse/httpapi.rst [hunk 4]
```diff
@@ -0,0 +1,804 @@
...
+                    buf += byts
+                    if not chunkend:
+                        continue
+
+                    mesg = json.loads(buf)
+                    buf = b""
+
+                    pprint.pprint(buf)
+
+/api/v1/storm/call
```
Comment (vEpiphyte) [inline docs/synapse/httpapi.rst:425]: This section is updated

File: docs/synapse/devguides/storm_api.rst [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+HTTP API
+--------
+
+The HTTP API versions of the Storm APIs can be found here `Cortex HTTP API`_.
+
+``/v1/api/storm``
+    This API returns a message stream.
+
+``/v1/api/storm/call``
+    This API returns a a message given by the Storm ``return( )`` syntax.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:39]: 'a a' message

File: docs/synapse/devguides/storm_api.rst [hunk 6]
```diff
@@ -0,0 +1,645 @@
...
+the messages that may be seen when consuming the message stream.
+
+Each message has the following basic structure::
+
+    [ "type", { ..type specific info... } ]
+
+init
+----
+
+First message sent by a storm query runtime.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:57]: should capitalize the 's' in 'Storm'

File: docs/synapse/devguides/storm_api.rst [hunk 7]
```diff
@@ -0,0 +1,645 @@
...
+It includes the following keys:
+
+task
+    The task identifier (which can be used for task cancellation).
+
+tick
+    The epoch time the query execution started (in milliseconds).
+
+text
+    The storm query text.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:68]: capitalize 'storm'

File: docs/synapse/devguides/storm_api.rst [hunk 8]
```diff
@@ -0,0 +1,645 @@
...
+This represents a packed node. Each serialized node will have the following structure::
+
+    [
+        [<form>, <valu>],       # The [ typename, typevalue ] definition of the node.
+        {
+            "iden": <hash>,     # A stable identifier for the node.
+            "tags": {},         # The tags on the node.
+            "props": {},        # The node's secondary properties.
+            "path": {},         # Path related information in the node.
+            "tagprops": {},     # The nodes tag properties.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:90]: missing apostrophe in 'nodes'

File: docs/synapse/devguides/storm_api.rst [hunk 9]
```diff
@@ -0,0 +1,645 @@
...
+Example::
+
+    (print, {'mesg': 'I am a message!'})
+
+This can be produced by users with the ``$lib.print()`` Storm API.
+
+warn
+----
+
+The warn event contains data about a non-fatal errors encountered when processing something.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:138]: extra 'a' in 'a non-fatal errors'

File: docs/synapse/devguides/storm_api.rst [hunk 10]
```diff
@@ -0,0 +1,645 @@
...
+eline
+    Line number from the raising file.
+
+ename
+    Name of the function where the exception was from.
+
+mesg
+    The ``mesg`` argument to a SynErr exception, if present; or the ``str()`` exception.
+
+If the Storm runtime is cancelled for some reason, there will will be no ``err`` or ``fini`` messages sent.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:182]: 'will will'

File: docs/synapse/devguides/storm_api.rst [hunk 11]
```diff
@@ -0,0 +1,645 @@
...
+count
+    The number of nodes yielded by the runtime.
+
+Example::
+
+    ('fini', {'count': 1, 'tock': 1539221715240, 'took': 36381})
+
+.. note::
+
+    If the Storm runtime is cancelled for some reason, there will will be no ``err`` or ``fini`` messages
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:223]: 'will will'

File: docs/synapse/devguides/storm_api.rst [hunk 12]
```diff
@@ -0,0 +1,645 @@
...
+look\:miss
+----------
+
+The ``look:miss`` message is sent when the Storm runtime is set to ``lookup`` mode and the node that was identified
+by the scrape logic is not present in the current View.
+
+It includes the following key:
+
+ndef
+    A tuple of the form and normalizxed value.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:308]: typo 'normalizxed'

File: docs/synapse/devguides/storm_api.rst [hunk 13]
```diff
@@ -0,0 +1,645 @@
...
+
+    # This query produces the following event: $lib.csv.emit(foo, bar, $lib.time.now(), table=foo)
+    ('csv:row', {'row': ('foo', 'bar', 1662578059282), 'table': 'foo'})
+
+.. _dev_storm_call:
+
+Storm Call APIS
+===============
+
+The Telepath ``callStorm()`` and HTTPAPI ``storm/call`` interfaces are designed to return a single message to the
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:343]: space in 'HTTP API' to be consistent with previous usage

File: docs/synapse/devguides/storm_api.rst [hunk 14]
```diff
@@ -0,0 +1,645 @@
...
+         'email': None,
+         'iden': '933a320b7ce8134ba5abd93aa487e1b5',
+         'locked': False,
+         'name': 'root',
+         'roles': (),
+         'rules': (),
+         'type': 'user'}
+
+
+    The following show setting an API key for a Power-Up. There is no ``return`` statement, so the return value
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:371]: add an 's' on 'show'

File: docs/synapse/devguides/storm_api.rst [hunk 15]
```diff
@@ -0,0 +1,645 @@
...
+----
+
+This is the mode that a Storm query is parsed in. This value can be specified to ``lookup``, ``autoadd``, and
+``search`` modes to get different behaviors.
+
+Example:
+
+    .. code:: python3
+
+        # Using lookup mode, the query text, before switching command mode with a | character,
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:458]: switching 'to'

File: docs/synapse/devguides/storm_api.rst [hunk 16]
```diff
@@ -0,0 +1,645 @@
...
+        # will have its text scrapped for simple values such as FQDNs, IP Addresses, and Hashes
+        # and attempt to lift any matching nodes.
+        opts = {'mode': 'lookup'}
+
+        # Using autoadds mode, the query text is scrapped like in lookup mode; and for any
+        # values  which we try to lift that do not produce nodes, those nodes will be added
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:464]: extra space between 'values  which'

File: docs/synapse/devguides/storm_api.rst [hunk 17]
```diff
@@ -0,0 +1,645 @@
...
+            ('foobar.com', '8.8.8.8', '20210810'),
+            ('bazplace.net', '1.2.3.4', '20210810'),
+        )
+        vars = {'records': records}
+        opts = {'vars': vars}
+
+view
+----
+
+The View iden in which to run the Storm query in. If not specified, the query will run in the users default view.
```
Comment (Cisphyx) [inline docs/synapse/devguides/storm_api.rst:630]: "user's"

File: docs/synapse/httpapi.rst [hunk 18]
```diff
@@ -0,0 +1,804 @@
...
+
+        # we are now clear to make additional HTTP API calls using sess
+
+/api/v1/active
+~~~~~~~~~~~~~~
+
+*Method*
+    GET
+
+    This is a unauthenticated API that returns the leader status of Cell.
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:105]: 'an' unauthenticated

File: docs/synapse/httpapi.rst [hunk 19]
```diff
@@ -0,0 +1,804 @@
...
+    *Returns*
+        A user dictionary.
+
+/api/v1/auth/password/<id>
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Method*
+    POST
+
+    This API allows the caller to change a users password. The authenticated user must either be an admin or
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:208]: "user's"

File: docs/synapse/httpapi.rst [hunk 20]
```diff
@@ -0,0 +1,804 @@
...
+
+    *Input*
+        This API expects the following JSON body::
+
+            { "name": "myuser" }
+
+        Any additional "user dictionary" fields (other than "iden") may be specified.
+
+    *Returns*
+        The newly created role dictionary.
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:144]: user dictionary

File: docs/synapse/httpapi.rst [hunk 21]
```diff
@@ -0,0 +1,804 @@
...
+    *Input*
+        This API expects a dictionary containing any updated values for the role.
+
+    *Returns*
+        The updated role dictionary.
+
+*Method*
+    GET
+
+    This API allows the caller to retrieve a user dictionary.
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:235]: role dictionary

File: docs/synapse/httpapi.rst [hunk 22]
```diff
@@ -0,0 +1,804 @@
...
+
+    *Returns*
+        A user dictionary.
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:238]: role dictionary

File: docs/synapse/httpapi.rst [hunk 23]
```diff
@@ -0,0 +1,804 @@
...
+            import pprint
+
+            # Assumes sess is an aiotthp client session that has previously logged in
+
+            query = '.created $lib.print($node.repr(".created")) | limit 3'
+            data = {'query': query, 'opts': {'repr': True}}
+            url = 'https://localhost:4443/api/v1/storm'
+
+            async with sess.get(url, json=data) as resp:
+                async for byts, x in resp.content.iter_chunks():
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:366]: should use the `chunkend` flag to reassemble the chunks in case they have been re-chunked by a proxy

Comment (Cisphyx) [reply docs/synapse/httpapi.rst:366]: nevermind i see you've got that later

File: docs/synapse/httpapi.rst [hunk 24]
```diff
@@ -0,0 +1,804 @@
...
+    GET
+    
+    *Returns*
+        If the file exists a stream of byte chunks will be returned to the caller. A ``Range`` header with a single ``bytes`` value can be provided to get a subset of a file.
+
+*Method*
+     HEAD
+     
+     *Returns*
+        If the file exists, the ``Content-Length`` header will be set for the size of the file. If a ``Range`` header with a single ``bytes`` value is provided, the ``Content-Length`` header will describe the size of hte range, and the ``Content-Range`` header will also be set to describe the range of the requested bytes.
```
Comment (Cisphyx) [inline docs/synapse/httpapi.rst:794]: 'hte' range

File: docs/synapse/httpapi.rst [hunk 25]
```diff
@@ -0,0 +1,804 @@
...
+
+For additional examples, see the code examples at `HTTPAPI Examples`_.
+
+Authentication
+--------------
+
+While not in "insecure" mode, most Synapse HTTP APIs require an authenticated user.
+HTTP API endpoints requiring authentication may be accessed using either HTTP Basic
+authentication via the HTTP "Authorization" header or as part of an authenticated
+session.  For more information on configuring users/roles see TODO.
```
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply docs/synapse/httpapi.rst]: removed

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**